examples/daemonsetsharding/deployment-no-node-pods.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: exporter
+    app.kubernetes.io/name: kube-state-metrics
+    app.kubernetes.io/version: 2.10.0
+  name: kube-state-metrics-pods
+  namespace: kube-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: kube-state-metrics
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: exporter
+        app.kubernetes.io/name: kube-state-metrics
+        app.kubernetes.io/version: 2.10.0
+    spec:
+      automountServiceAccountToken: true
+      containers:
+      - args:
+        - --resources=pods
+        - --node=""
+        - --enable-no-node-scrape
```

Comment:
I think we can remove `- --enable-no-node-scrape`. If node == "", then it will collect pod metrics whose don't have  assigned nodes.

Reply:
QQ: could you provide a config example on `only namespace is selected regardless of whether it is assigned to node.`?

Is it --node="" && --namespace="kube-system"

Reply:
Yes, let me run a test to confirm.

Reply:
I just ran a test with this PR version.
My cluster currently has the following pods pending
<img width="818" alt="image" src="https://github.com/kubernetes/kube-state-metrics/assets/1990362/5be7f069-84c2-42a7-a251-7cdcaa287893">
When I use the following configuration，it can scrape to pending pods under namespace media
```
          args:
            - '--resources=pods'
            - '--node='
            - '--enable-no-node-scrape'
            - '--namespaces=media'
```
When I use the following configuration，it can scrape to all pods under namespace media
```
          args:
            - '--resources=pods'
            - '--node='
            - '--namespaces=media'
```

Reply:
We need to change this:

```
func (n *NodeType) GetNodeFieldSelector() string {
	if string(*n) != "" {
		return fields.OneTermEqualSelector("spec.nodeName", string(*n)).String()
	}
	return EmptyFieldSelector()
}
```
to 
```
func (n *NodeType) GetNodeFieldSelector() string {
	return fields.OneTermEqualSelector("spec.nodeName", string(*n)).String()
}
```

https://github.com/kubernetes/kube-state-metrics/blob/5c71459a614eaf8370bed2f406899f2269edff66/pkg/options/types.go#L111C20-L116

Reply:
Oh, I realized the issue.

If we change the above, it will break for `--namespaces=media`. Because we assume collecting all nodes when node = "".

Reply:
Yes, it is. I think the current PR version is flexible. So this configuration combination was added.

Reply:
One solution is to 1. change node type from string to list 2. add an assert that #nodes <= 1:

```
o.cmd.Flags().StringVar((*string)(&o.Node), "node"
```
to 
```
o.cmd.Flags().Var(&o.Namespaces, "namespaces", # similar to namespaces.
```
So that we can differentiate between nil and "". 

https://github.com/kubernetes/kube-state-metrics/blob/5c71459a614eaf8370bed2f406899f2269edff66/pkg/options/options.go#L139C2-L139C2

Reply:
QQ: what happened during upgrade? I thought it only had one small breaking change:

Before: --node= works
Now: --node= works

One breaking change: 
Before: --node="" means collecting all nodes.
Now: --node="" means collecting empty node.

But I think most people will not set --node if they don't use this field. We can update release note about it. Since it's an experimental feature, it could have this fix.

I think this way is unified. I didn't realize collecting empty node metrics, and we should fix it.

Reply:
I'm testing the code, just a little confused to make sure. I will submit the revised version later.